activation/challenge_verifier.go

```diff
@@ -0,0 +1,155 @@
+package activation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	atypes "github.com/spacemeshos/go-spacemesh/activation/types"
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/log"
+	"github.com/spacemeshos/go-spacemesh/signing"
+)
+
+var (
+	ErrSignatureInvalid = errors.New("signature is invalid")
+	ErrChallengeInvalid = errors.New("challenge is invalid")
+)
+
+type CouldntVerifyError struct {
```

Comment:
it should suffice to use fmt.errorf for this and ATXNotFound too.
i don't see the need to create two custom errors.